lib/vagrant-service-manager/command.rb

```diff
@@ -266,21 +266,24 @@ def print_vagrant_box_version(script_readable = false)
       end
 
       def restart_service(service)
-
-        if SCCLI_SERVICES.include? service
-          # TODO : Handle the case where user wants to pass extra arguments to OpenShift service
-          command = "sudo sccli #{service}"
-        else
-          command = "sudo systemctl restart #{service}"
-        end
-
-        with_target_vms(nil, { single_target: true}) do |machine|
-          machine.communicate.execute(command) do |type, error|
-            if type == :stderr
-              @env.ui.error(error)
-              exit 126
-            end
+        errors = []
+        command = if SCCLI_SERVICES.include? service
+                    # TODO : Handle the case where user wants to pass extra arguments
+                    # to OpenShift service
+                    "sccli #{service}"
+                  else
+                    "systemctl restart #{service}"
+                   end
+
+        with_target_vms(nil, single_target: true) do |machine|
+          exit_code = machine.communicate.sudo(command) do |type, error|
+            errors << error if type == :stderr
+          end
+          unless exit_code.zero?
+            @env.ui.error errors.join("\n")
+            exit exit_code
           end
+          exit_code
```

Comment:
Although we do not need to explicitly return exit_code here (whose value would be `0`) as this will be returned by default, but okay!